src/system.vala

```diff
@@ -208,20 +221,44 @@ namespace Frida {
 		private TemporaryDirectory directory;
 
 		public TemporaryFile.from_stream (string name, InputStream istream, TemporaryDirectory? directory = null) throws Error {
-			if (directory != null)
+			int memfd_fd = -1;
+
+			if (directory != null) {
 				this.directory = directory;
-			else
-				this.directory = TemporaryDirectory.system_default;
-			this.file = File.new_for_path (Path.build_filename (this.directory.path, name));
+				this.file = File.new_for_path (Path.build_filename (this.directory.path, name));
+			}
+			else {
+				memfd_fd = System.memfd_create(name, ALLOW_SEALING);
+				//  memfd_fd = -1;
+				if(memfd_fd >= 0) {
+					this.file = File.new_for_path (Path.build_filename ("/proc", ((int)Posix.getpid()).to_string("%d"), "fd", memfd_fd.to_string("%d")));
+					this.directory = new TemporaryDirectory.with_file (this.file, false);
+				}
+				else {
+					this.directory = TemporaryDirectory.system_default;
+					this.file = File.new_for_path (Path.build_filename (this.directory.path, name));
+				}
+			}
 
 			try {
 				// FIXME: REPLACE_DESTINATION doesn't work?!
-				file.delete ();
+				if(memfd_fd < 0) {
+					file.delete ();
+				}
 			} catch (GLib.Error delete_error) {
 			}
 
 			try {
-				var ostream = file.create (FileCreateFlags.REPLACE_DESTINATION, null);
+				IOStream iostream;
+				OutputStream ostream;
+				if(memfd_fd >= 0) {
+					iostream = file.open_readwrite (null);
+					ostream = iostream.output_stream;
+				}
+				else {
+					ostream = file.create (FileCreateFlags.REPLACE_DESTINATION, null);
+				}
+				
```

Comment:
For consistency there should never be more than one blank line used to separate code, here there's two.